core/arch/arm/kernel/spmc_sp_handler.c

```diff
@@ -1106,6 +1106,47 @@ static void spmc_handle_version(struct thread_smc_args *args,
 		      FFA_PARAM_MBZ, FFA_PARAM_MBZ);
 }
 
+static void handle_console_log(struct thread_smc_args *args)
+{
+	uint32_t ret_fid = FFA_ERROR;
+	uint32_t ret_val = FFA_INVALID_PARAMETERS;
+	size_t char_count = (args->a1 & FFA_CONSOLE_LOG_CHAR_COUNT_MASK);
+	const void *reg_list[] = {
+		&args->a2, &args->a3, &args->a4,
+		&args->a5, &args->a6, &args->a7
+	};
+	char buffer[FFA_CONSOLE_LOG_64_MAX_MSG_LEN + 1];
+	size_t max_length = 0;
+	size_t reg_size = 0;
+	size_t n = 0;
+
+	if (args->a0 == FFA_CONSOLE_LOG_64) {
+		max_length = FFA_CONSOLE_LOG_64_MAX_MSG_LEN;
+		reg_size = sizeof(uint64_t);
+	} else {
+		max_length = FFA_CONSOLE_LOG_32_MAX_MSG_LEN;
+		reg_size = sizeof(uint32_t);
+	}
+
+	if (char_count < 1 || char_count > max_length)
+		goto out;
+
+	for (n = 0; n < char_count; n += reg_size)
+		memcpy(buffer + n, reg_list[n / reg_size],
+		       MIN((char_count - n), reg_size));
```

Comment:
Nit: the `()` around `char_count - n` are not needed.

Reply:
Done